File: 04/server/message.js
class Message {
    constructor(msg_type, action, academics_codes,
        course_code, academic_year, academic_semester,
        grade, absences) {
        
        this.msg_type = msg_type
        this.action = action
        this.academics_codes = academics_codes
        this.course_code = course_code
        this.academic_year = academic_year
        this.academic_semester = academic_semester
        this.grade = grade
        this.absences = absences
    }

    get_msg_type = () => {
        return this.msg_type
    }

    set_msg_type = (value) => {
        this.msg_type = value
    }

    get_action = () => {
        return this.action
    }

    set_action = (value) => {
        this.action = value
    }
    
    get_academics_codes = () => {
        return this.academics_codes
    }

    set_academics_codes = (value) => {
        this.academics_codes = value
    }
    
    get_course_code = () => {
        return this.course_code
    }

    set_course_code = (value) => {
        this.course_code = value
    }
    
    get_academic_year = () => {
        return this.academic_year
    }

    set_academic_year = (value) => {
        this.academic_year = value
    }
    
    get_academic_semester = () => {
        return this.academic_semester
    }

    set_academic_semester = (value) => {
        this.academic_semester = value
    }
    
    get_grade = () => {
        return this.grade
    }

    set_grade = (value) => {
        this.grade = value
    }
    
    get_absences = () => {
        return this.absences
    }

    set_absences = (value) => {
        this.absences = value
    }

}

File: 04/client/message.py
class Message:
    """Class representing a message"""
    
    msg_type = None  # 1 - request or 2 - response
    action = None  # 1 - set, 2 - remove or 3 - query
    academics_codes = None  # students ids
    course_code = None  # course id
    academic_year = None  # academic year
    academic_semester = None  # academic semester
    grade = None  # grade to set
    absences = None  # absences to set

    def __init__(self, msg_type=1, action=3,
                 academics_codes=[], course_code=None,
                 academic_year=None, academic_semester=None,
                 grade=None, absences=None):
        self.msg_type = msg_type
        self.action = action
        self.academics_codes = academics_codes
        self.course_code = course_code
        self.academic_year = academic_year
        self.academic_semester = academic_semester
        self.grade = grade
        self.absences = absences

    def get_msg_type(self):
        return self.msg_type

    def set_msg_type(self, value):
        self.msg_type = value

    def get_action(self):
        return self.action

    def set_action(self, value):
        self.action = value
    
    def get_academics_codes(self):
        return self.academics_codes

    def set_academics_codes(self, value):
        self.academics_codes = value
    
    def get_course_code(self):
        return self.course_code

    def set_course_code(self, value):
        self.course_code = value
    
    def get_academic_year(self):
        return self.academic_year

    def set_academic_year(self, value):
        self.academic_year = value
    
    def get_academic_semester(self):
        return self.academic_semester

    def set_academic_semester(self, value):
        self.academic_semester = value
    
    def get_grade(self):
        return self.grade

    def set_grade(self, value):
        self.grade = value
    
    def get_absences(self):
        return self.absences

    def set_absences(self, value):
        self.absences = value